### PR TITLE
Cu 86b0qp555 implement protected routes jose

### PR DESCRIPTION
**Pull request by josewebdev2000**
When the web app fired first before, the user was directed to the dashboard of the app instead of the login page. Now, when the application fires up, the user is redirected to the login page. Also, when the user logs out he is redirected to the login page.

Hence, unauthenticated users cannot use the web app.

### CODE REVIEW

**Comment by allantito** [comment]
Task linked: [CU-86b0qp555 Implement protected routes](https://app.clickup.com/t/86b0qp555)

File: P1/Project_Code/frontend/src/components/Layout/Layout.jsx [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+import { useContext, useEffect }from "react";
+import { Outlet, useNavigate } from "react-router-dom";
 import { NavigationBar } from "../Navigation/NavigationBar.jsx";
 import { SearchBar } from "../Search/SearchBar.jsx";
 import { Grid } from "@mui/material";
 
+import { UserContext } from "../../context/User.context.jsx";
+
 export const Layout = () => {
+
```
Comment (ethanfreire) [inline P1/Project_Code/frontend/src/components/Layout/Layout.jsx:10]: Good use of comments here

File: P1/Project_Code/frontend/src/routes/Login/Login.route.jsx [hunk 2]
```diff
@@ -48,24 +41,50 @@ function Login() {
 		}
 	}
 
+	//Error validation function
+	const validateInput = async (data) => {
+		let inputErrors = {};
+
+		if (!isEmail(data.email))
```
Comment (ethanfreire) [inline P1/Project_Code/frontend/src/routes/Login/Login.route.jsx:48]: Consider adding a check that the value of data.email and data.password are not null or undefined. See example below 
```
if (!data.email || !isEmail(data.email)) {
      inputErrors.email = "Invalid Email";
}
```

File: P1/Project_Code/frontend/src/routes/Register/Register.route.jsx [hunk 3]
```diff
@@ -57,6 +49,11 @@ function Register() {
 	const validateInput = async (data) => {
 		let inputErrors = {};
 
+		if (!isEmail(data.email))
```
Comment (ethanfreire) [inline P1/Project_Code/frontend/src/routes/Register/Register.route.jsx:52]: Same as my last comment add an explicit check that data.email values are not undefined or null.

**Review by ethanfreire — COMMENTED** [review]
I've left some comment for your to review @josewebdev2000

**Pull request merged**